src/thunderbots/software/gui/widgets/main_widget.cpp

```diff
@@ -0,0 +1,494 @@
+#include "software/gui/widgets/main_widget.h"
+
+// TODO: comment. this is the autogenerated file from the ui file
```

Comment:
TODO? Also can we generate this as part of the build? Ideally we don't commit generated code at all if we can avoid it.

Reply:
Also is this autogenerated? Seems like there are a lot of comments and whatnot here that look like you wrote them?

Reply:
Gonna hold off reviewing this file until you let me know either way.....

Reply:
Sorry this file is not autogenerated, the comment was referring to the file being included.